src/XYZArray.h

```diff
@@ -40,6 +40,18 @@ class XYZArray
     Init(n);
   }
 
+  //Create a struct of arrays from an array of structs
+  XYZArray(std::vector<XYZ> AOS)
```

Comment:
Pass `AOS` by reference. This is invoking copy instructor!

Reply:
Fixed.